### PR TITLE
Add defer and stream support to supermassive

### PR DESCRIPTION
**Pull request by freiksenet**

### CODE REVIEW

File: packages/apollo-mock-client/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "just": "monorepo-scripts"
   },
   "devDependencies": {
-    "@apollo/client": "^3.3.15",
+    "@apollo/client": "3.7.0",
```
Comment (alloy) [inline packages/apollo-mock-client/package.json:20]: Why are you doing this?

File: .azure-devops/graphitation-release.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 pr: none
 trigger:
   - main
+  - supermassive-v3
```
Comment (alloy) [inline .azure-devops/graphitation-release.yml:4]: So I believe this will need to be merged into main

File: packages/supermassive/.eslintrc.json [hunk 3]
```diff
@@ -1,4 +1,23 @@
...
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "rules": {
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": [
+      "error",
+      { "argsIgnorePattern": "^_" }
+    ]
+  },
```
Comment (alloy) [inline packages/supermassive/.eslintrc.json:13]: What's different here?

File: packages/supermassive/.eslintrc.json [hunk 4]
```diff
@@ -1,4 +1,23 @@
...
+  "overrides": [
+    {
+      "files": ["**/__tests__/**"],
+      "rules": {
+        "@typescript-eslint/no-explicit-any": "off"
+      }
+    }
```
Comment (alloy) [inline packages/supermassive/.eslintrc.json:20]: Perhaps we want this as a root option?

File: tsconfig.json [hunk 5]
```diff
@@ -9,7 +9,8 @@
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
     "paths": {
-      "@graphql-tools/utils": ["./node_modules/@graphql-tools/utils"]
+      "@graphql-tools/utils": ["./node_modules/@graphql-tools/utils"],
+      "graphql": ["./packages/supermassive/node_modules/graphql"]
```
Comment (alloy) [inline tsconfig.json:13]: So we're using latest version for typings for all monorepo packages, but not for actual runtime?

**Review by alloy — COMMENTED** [review]
Nothing stands out, looks good thus far. Are there any specific supermassive differences in implementation that you could point out?

File: .azure-devops/graphitation-release.yml (hunk 2 above)
Comment (freiksenet) [reply .azure-devops/graphitation-release.yml:4]: Good point, thanks.

File: packages/apollo-mock-client/package.json (hunk 1 above)
Comment (freiksenet) [reply packages/apollo-mock-client/package.json:20]: Your stuff breaks with apoll 3.7.15.

File: packages/supermassive/.eslintrc.json (hunk 3 above)
Comment (freiksenet) [reply packages/supermassive/.eslintrc.json:13]: That is a merge artifact, I did eslint there first. Will fix.

File: packages/supermassive/.eslintrc.json (hunk 4 above)
Comment (freiksenet) [reply packages/supermassive/.eslintrc.json:20]: Same as above.

File: tsconfig.json (hunk 5 above)
Comment (freiksenet) [reply tsconfig.json:13]: Yeah, that was the only way I managed to get it to run.

File: packages/supermassive/src/__tests__/execute.test.ts [hunk 6]
```diff
@@ -295,73 +313,505 @@ query Person($id: Int!) {
...
+      gender
+    }
+  }
+ `,
+    variables: {
+      limit: 5,
+      throwError: true,
+    },
+  },
+  // defer
```
Comment (vladar) [inline packages/supermassive/src/__tests__/execute.test.ts:360]: There are a few cases where behavior is not quite clear from those tests:

1. Fields that exist inside both - normal selection set, and deferred one (especially nested ones):

```graphql
query {
  person {
    account { signupDate }
    ...PersonDetails @defer
   }
}

fragment PersonDetails on Person {
  account {
    __typename
    signupDate
  }
}
```
(same probably works with streams - especially deeply nested ones)

3. Include/skip inside defer/stream

4. Interplay between defer/stream and include/skip in various combinations, e.g.
```graphql
query($include: Boolean = false) {
  person {
    account { signupDate }
    ...PersonDetails @include(if: $include) @defer
   }
}

fragment PersonDetails on Person {
  account {
    __typename
    signupDate
  }
}
```

5. Error bubbling inside defer/stream

Comment (vladar) [reply packages/supermassive/src/__tests__/execute.test.ts:360]: 5. `defer` for fragment on lists:

```graphql
query {
  someList {
     ...MyListItem @defer
  }
}
fragment MyListItem on ListItem {
  foo
}
```

6. `defer` for fragment on lists of union type:
```graphql
query {
  someListOnUnion {
     ...MyFoo @defer
     ...MyBar
  }
}
fragment MyFoo on UnionMemberFoo {
  foo
}
fragment MyBar on UnionMemberBar {
  bar
}
```

Comment (vladar) [reply packages/supermassive/src/__tests__/execute.test.ts:360]: Different combinations of fragments on abstract types could be tricky (e.g. fragment on interface not deferred, fragment on specific - deferred and vice-versa

**Pull request merged**